### ISSUE TITLE
Entity diagram of database classes

### ISSUE THREAD

**Issue by mmnosek** [Starcounter/Home#147]
It would be nice to be able to generate some overall view of the whole datamodel. I can imagine two ways:

1. Generate it out of database based on currently loaded apps
2. Generate it out of C# code (in VS)

Do we have anything like that (I guess not option 1) but has anybody tried option 2?

cc @Starcounter/sc-apps @TakhirMamirov

**Comment by per-samuelsson** [comment]
> Do we have anything like that (I guess not option 1) but has anybody tried option 2?

No we don't. There is some "database editor" app @andwah prototyped. Maybe that can be usable.

Furthermore, you can retreive schema-like content (relevant to database classes) from the weaver, based on an app:

```
scweaver schema path/to/your/app.exe
```

but it's unpolished and never promoted (basically just used by me internally I think).

**Comment by ubbeK** [comment]
> scweaver schema path/to/your/app.exe

Should this work with starcounter 2.3.1? I only get that it cannot find any schema when trying it out on our HelloWorld.

**Comment by per-samuelsson** [comment]
> Should this work with starcounter 2.3.1?

Yes, but I forgot to mention, that in 2.3, we weave start-time, hence you need to explicitly weave it first:

```
scweaver HelloWorld.exe
scweaver schema HelloWorld.exe
```

Sorry.

WTBS, this is not a production-like, promoted thing, like I wrote before. Just a hint about something we have today that could be worth knowing about.

**Comment by ubbeK** [comment]
@per-samuelsson I understand, would just like to try it out for myself.

**Comment by k-rus** [comment]
It is also possible to query metadata to get to know about loaded database classes and their properties.

**Comment by mmnosek** [comment]
> No we don't. There is some "database editor" app @andwah prototyped. Maybe that can be usable.

@andwah any clue on this? 

Actually I think that problem of generating class diagram out of C# code isn't Starcounter-specific, so any other tools like VS Ultimate or Doxygen might be used. The problem starts when we think about the structure of the database itself. These tools have no idea about which classes are persistent and which are not.

**Comment by warpech** [comment]
That's the app @per-samuelsson referred to: https://github.com/Smorgasbord-Development/DatabaseEditor
